tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile

```diff
@@ -27,11 +27,12 @@ RUN sh -c "echo '${IMG_USER} ALL=NOPASSWD: ALL' >> /etc/sudoers"
 
 #FIXME: gcc is required as agent is build out of a container build.
 RUN apt-get update && \
-  apt install -y \
-  gcc \
-  git \
-  make \
-  xz-utils
+    apt install --no-install-recommends -y \
+    gcc \
+    git \
+    make \
+    xz-utils \
+    cpio
```

Comment:
I don't think adding `cpio` will do what we're expecting, or fix the #3144.

Let's do a local test here, using `ubuntu` (which we actually should tag to a version, rather than using the latest, but that's for a different issue).
```sh
[fidencio@kundera kata-containers]$ docker run -it ubuntu
root@2fb1bda30d41:/# apt install cpio
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package cpio is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  tar

E: Package 'cpio' has no installation candidate
```

And tar already comes as part of the container image:
```sh
[fidencio@kundera kata-containers]$ docker run -it ubuntu
root@efa2079d57bb:/# dpkg -l tar
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version                    Architecture Description
+++-==============-==========================-============-========================================
ii  tar            1.30+dfsg-7ubuntu0.20.04.1 amd64        GNU version of the tar archiving utility
```

So, adding `cpio` should actually fail, but it doesn't as `--no-install-recommends -y` is passed.

All in all, we shouldn't add the `cpio` as it's already part of the `tar` package, which is already present as part of the ubuntu container image.

Reply:
Oops, scratch everything I said, I'm 100% mistaken. :-)
```sh
[fidencio@kundera kata-containers]$ docker run -it ubuntu
root@a41ba96dea79:/# apt update
Get:1 http://security.ubuntu.com/ubuntu focal-security InRelease [114 kB]
Get:2 http://archive.ubuntu.com/ubuntu focal InRelease [265 kB]
Get:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]
Get:4 http://archive.ubuntu.com/ubuntu focal-backports InRelease [101 kB]
Get:5 http://security.ubuntu.com/ubuntu focal-security/universe amd64 Packages [821 kB]
Get:6 http://archive.ubuntu.com/ubuntu focal/universe amd64 Packages [11.3 MB]
Get:7 http://security.ubuntu.com/ubuntu focal-security/main amd64 Packages [1281 kB]
Get:8 http://security.ubuntu.com/ubuntu focal-security/restricted amd64 Packages [682 kB]
Get:9 http://security.ubuntu.com/ubuntu focal-security/multiverse amd64 Packages [30.1 kB]
Get:10 http://archive.ubuntu.com/ubuntu focal/main amd64 Packages [1275 kB]                
Get:11 http://archive.ubuntu.com/ubuntu focal/restricted amd64 Packages [33.4 kB]
Get:12 http://archive.ubuntu.com/ubuntu focal/multiverse amd64 Packages [177 kB]
Get:13 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages [1098 kB]
Get:14 http://archive.ubuntu.com/ubuntu focal-updates/restricted amd64 Packages [738 kB]
Get:15 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [1695 kB]
Get:16 http://archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 Packages [33.6 kB]
Get:17 http://archive.ubuntu.com/ubuntu focal-backports/main amd64 Packages [2668 B]
Get:18 http://archive.ubuntu.com/ubuntu focal-backports/universe amd64 Packages [7185 B]
Fetched 19.8 MB in 2s (10.9 MB/s)                         
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.
root@a41ba96dea79:/# apt install cpio

root@a41ba96dea79:/# apt install cpio
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Suggested packages:
  libarchive1
The following NEW packages will be installed:
  cpio
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 86.4 kB of archives.
After this operation, 377 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 cpio amd64 2.13+dfsg-2ubuntu0.3 [86.4 kB]
Fetched 86.4 kB in 0s (473 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package cpio.
(Reading database ... 4127 files and directories currently installed.)
Preparing to unpack .../cpio_2.13+dfsg-2ubuntu0.3_amd64.deb ...
Unpacking cpio (2.13+dfsg-2ubuntu0.3) ...
Setting up cpio (2.13+dfsg-2ubuntu0.3) ...
update-alternatives: using /bin/mt-gnu to provide /bin/mt (mt) in auto mode
update-alternatives: warning: skip creation of /usr/share/man/man1/mt.1.gz because associated file /usr/share/man/man1/mt-gnu.1.gz (of link group mt) doesn't exist
```

It works, and it shows that I'm totally not familiar with ubuntu, as I didn't know one would have to `apt update` before actually being able to install a pacakge.

Removing the `do-not-merge` label and approving the PR.

Reply:
@fidencio Sorry to bother you , but I think you should execute `apt-get update` as first in the ubuntu container. Then you will find `cpio` package.
```sh
❯ docker run --rm -it --name test ubuntu:20.04 bash
root@f8fdac4eb940:/# apt-get update
Get:1 http://archive.ubuntu.com/ubuntu focal InRelease [265 kB]
...
Get:18 http://archive.ubuntu.com/ubuntu focal-backports/main amd64 Packages [2668 B]
Fetched 19.8 MB in 15s (1362 kB/s)
Reading package lists... Done
root@f8fdac4eb940:/# apt-get install --no-install-recommends -y cpio
Reading package lists... Done
Building dependency tree
Reading state information... Done
Suggested packages:
  libarchive1
The following NEW packages will be installed:
  cpio
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 86.4 kB of archives.
After this operation, 377 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 cpio amd64 2.13+dfsg-2ubuntu0.3 [86.4 kB]
Fetched 86.4 kB in 2s (55.9 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package cpio.
(Reading database ... 4127 files and directories currently installed.)
Preparing to unpack .../cpio_2.13+dfsg-2ubuntu0.3_amd64.deb ...
Unpacking cpio (2.13+dfsg-2ubuntu0.3) ...
Setting up cpio (2.13+dfsg-2ubuntu0.3) ...
update-alternatives: using /bin/mt-gnu to provide /bin/mt (mt) in auto mode
update-alternatives: warning: skip creation of /usr/share/man/man1/mt.1.gz because associated file /usr/share/man/man1/mt-gnu.1.gz (of link group mt) doesn't exist
root@f8fdac4eb940:/# cpio --help
Usage: cpio [OPTION...] [destination-directory]
GNU `cpio' copies files to and from archives
```
And thank you for reminding me that we need use the image with a tag(not latest), also the ci failure should be related to the script with error checking, now it is `build image` failure, I am checking and considering fixing.

Reply:
Yep, I noticed my mistake after I had already commented /o\.

I am building it locally here and checking whether it actually produces a valid initrd, have you done the check locally?

Reply:
Okay, just tested it locally with your patch, and it works as expected!